registry/auth.go

```diff
@@ -74,6 +74,12 @@ func (auth *RequestAuthorization) getToken() (string, error) {
 	return "", nil
 }
 
+// Checks that requests to the v2 registry can be authorized.
+func (auth *RequestAuthorization) IsAuthorized() bool {
```

Comment:
yeah I am digging through to see, since on push the first request is often sending data so we cannot rely on the HEAD check to the layer. A head on the manifest could be a solution but introduces an extra request.